notes.go

```diff
@@ -136,7 +136,33 @@ func (srv *notesAPI) DeleteNote(ctx context.Context, req *notesv1.DeleteNoteRequ
 }
 
 func (srv *notesAPI) ListNotes(ctx context.Context, req *notesv1.ListNotesRequest) (*notesv1.ListNotesResponse, error) {
-	return nil, status.Error(codes.Unimplemented, "unimplemented")
+	token, err := srv.authenticate(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	err = validators.ValidateListNoteRequest(req)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	// Check user is part of the group.
+	_, err = srv.groups.GetGroup(ctx, &models.OneGroupFilter{GroupID: req.GroupId}, token.AccountID)
+	if err != nil {
+		return nil, statusFromModelError(err)
+	}
```

Comment:
The `req.GroupId` field is optional. One can either list all notes that belong to a group or all notes that belong to a user (across many groups) or both.

You should rework the endpoint's logic so that it supports

```
ListNotes(group_id)
ListNotes(group_id, author_account_id)
ListNotes(author_account_id)
```